lms/djangoapps/grades/tests/integration/test_events.py

```diff
@@ -0,0 +1,126 @@
+"""
+Test grading event across apps.
+"""
+# pylint: disable=protected-access
+
+import json
+
+from capa.tests.response_xml_factory import MultipleChoiceResponseXMLFactory
+from lms.djangoapps.course_blocks.api import get_course_blocks
+from lms.djangoapps.instructor.enrollment import reset_student_attempts
+from mock import patch
+from openedx.core.djangolib.testing.utils import get_mock_request
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+from courseware.models import StudentModule
+from courseware.tests.test_submitting_problems import ProblemSubmissionTestMixin
+from student.models import CourseEnrollment
+from student.tests.factories import UserFactory
+from ...new.subsection_grade import SubsectionGradeFactory
+
+
+class DeleteStateEventIntegrationTest(ProblemSubmissionTestMixin, SharedModuleStoreTestCase):
+    """
+    Tests integration between the eventing in various layers
+    of the grading infrastructure.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(DeleteStateEventIntegrationTest, cls).setUpClass()
+        cls.course = CourseFactory.create()
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            category="chapter",
+            display_name="Test Chapter"
+        )
+        cls.sequence = ItemFactory.create(
+            parent=cls.chapter,
+            category='sequential',
+            display_name="Test Sequential 1",
+            graded=True,
+            format="Homework"
+        )
+        cls.vertical = ItemFactory.create(
+            parent=cls.sequence,
+            category='vertical',
+            display_name='Test Vertical 1'
+        )
+        problem_xml = MultipleChoiceResponseXMLFactory().build_xml(
+            question_text='The correct answer is Choice 3',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+        cls.problem = ItemFactory.create(
+            parent=cls.vertical,
+            category="problem",
+            display_name="Test Problem",
+            data=problem_xml
+        )
+
+    def setUp(self):
+        super(DeleteStateEventIntegrationTest, self).setUp()
+        self.request = get_mock_request(UserFactory())
+        self.student = self.request.user
+        self.client.login(username=self.request.user.username, password="test")
```

Comment:
Nitty: use `self.student` henceforth in this method instead of `self.request.user` as the former is more semantically understood.